### DISCUSSION TITLE
Query on document ID from index?

### DISCUSSION THREAD

**Discussion by mcrio** [ravendb/ravendb#18204] [Q&A] [answered]
Hi,

When creating an index there will always be an auto-generated `id()` term. 

Using the C# client query is there a way to filter on that term? A workaround would be to include an `Id` property into the index result class but that would be duplication.

Pseudo:

```
class Shirt {
  string Id;
  string Color;
}

class IndexResult {
  string Color;
}

query = session.query<indexResult, index>();
if (excludeDocumentId) {
  query = query.Where(item => item.Id != excludeDocumentId)  << How to do that without having Id on the index result class?
}
return query.where(item => item.Color == color);
```

**Comment by ayende** [comment, marked as answer]
Using the Linq in provider, not by default. You can use `RegisterCustomQueryTranslator` to add a convention based on your needs, though.

**Reply by adambajguz** [reply]
Hi @ayende can you please give more info on how to do it?

**Reply by ayende** [reply]
What _exactly_ is it that you are trying to do?
Given that you aren't the original poster, and the time since this thread started, it would be best to create a _new_ thread.

**Reply by adambajguz** [reply]
I noticed this thread and I'm simply curious how it can be done? There are not many examples of `RegisterCustomQueryTranslator` and the official docs doesn't even mention it (at least I could find it).

**Reply by ayende** [reply]
There are two sides of that. You need to define an operation (operator or method call) and you register that in the system.
There are several ways to customize the query translation process in RavenDB, which is why I asked for the actual scenario, to know which one to recommend.

That said, it is usually better to just go with the defaults.
